collect_app/src/test/java/org/odk/collect/android/formmanagement/ServerFormsDetailsFetcherTest.java

```diff
@@ -167,6 +172,22 @@ public void whenAFormExists_andItsNewerVersionWithUpdatedMediaFilesHasBeenAlread
         assertThat(form.isNotOnDevice(), is(false));
     }
 
+    @Test
+    public void whenAFormExists_andItsNewerVersionIsAvailableButHasNullHash_isNotNewOrUpdated() throws Exception {
```

Comment:
I'm just trying to think through this behaviour and if there are any problems in it being the case. Before this situation would just crash right?

Reply:
Yes this test was failing with the same stacktrace like we can see in firebase reports.